src/main/java/com/metamx/emitter/core/factory/EmitterFactory.java

```diff
@@ -0,0 +1,11 @@
+package com.metamx.emitter.core.factory;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.metamx.common.lifecycle.Lifecycle;
+import com.metamx.emitter.core.Emitter;
+import com.metamx.http.client.HttpClient;
+
+public interface EmitterFactory
```

Comment:
I have not finished it, but does https://github.com/metamx/emitter/pull/25 make sense?